pyblish/util.py

```diff
@@ -73,16 +126,22 @@ def publish(context=None, plugins=None, targets=None):
     }
 
     # Second pass, the remainder
+    tasks = list(logic.Iterator(plugins, context, state))
     for Plugin, instance in logic.Iterator(plugins, context, state):
```

Comment:
You can use the tasks you compute here, `for Plugin, instance in tasks:`

Reply:
I actually tried that, but some tests fail:

```
======================================================================
FAIL: Decrementing order works fine
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\tje\AppData\Local\Continuum\Miniconda2\envs\tgbvfx-environment\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\Users\tje\Documents\conda-git-deployment\repositories\tgbvfx-environment\pyblish-base\tests\pre13\test_logic.py", line 359, in test_decrementing_order
    assert_equals(count["#"], 111.1)
AssertionError: 1111.1 != 111.1
-------------------- >> begin captured stdout << ---------------------
I will run first, and stop things
I will stop things
I will not run

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: The test halts an invalid publish
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\tje\AppData\Local\Continuum\Miniconda2\envs\tgbvfx-environment\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\Users\tje\Documents\conda-git-deployment\repositories\tgbvfx-environment\pyblish-base\tests\pre13\test_logic.py", line 382, in test_test
    assert count["#"] == 11
AssertionError:
-------------------- >> begin captured stdout << ---------------------
I will run first, and stop things
I will not run

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: ContextPlugin works as advertised
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\tje\AppData\Local\Continuum\Miniconda2\envs\tgbvfx-environment\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\Users\tje\Documents\conda-git-deployment\repositories\tgbvfx-environment\pyblish-base\tests\test_plugin.py", line 563, in test_explicit_plugin
    assert count["#"] == 111, count
AssertionError: {'#': 1111}
-------------------- >> begin captured stdout << ---------------------
fail

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
pyblish.Collector: INFO: Collecting from ContextPlugin
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
```

I don't quite understand it, but I think its because the iterator raises an exception https://github.com/pyblish/pyblish-base/blob/477f540981433867aa219cc46cd0fa5890ecf17c/pyblish/logic.py#L372

Reply:
Ah, yes you are right.

This is another thing about iterators, while running, you can change the conditions within which it considers the next item to yield. In this case, your percentage won't be accurate. It may be "close enough", you'll have to experiment.

Reply:
> It may be "close enough", you'll have to experiment.

I might not be understanding the processing flow correctly, but in order to get an accurate percentage I would have to predict the flow which means running the ```process``` methods twice?
I don't really mind that the percentage isn't fully accurate. You could also argue that if a validator fails, the progress bar will stop. Dunno if that is the same scenario.